### DISCUSSION TITLE
Improved Media Manager

### DISCUSSION THREAD

**Discussion by bembelimen** [joomla/joomla-cms#31414] [2. Cold Features]
### Is your feature request related to a problem? Please describe.
With Joomla! 4.0, a completely rewritten and better version of the media manager was implemented.The major feature of Joomla 4.1 will be an even more improved media manager. Right now the media manager is mainly used for uploading images, but we want to extend it to a true media managing facility. Consider image management with different sizes and metadata, handling and specifically playback of video and audio and support for documents.

### Additional context

This Issue is a reminder for Joomla! 4.1: https://www.joomla.org/announcements/release-news/5819-the-day-after-joomla-4-0-planning-the-future.html

First implementations: https://github.com/bembelimen/joomla-cms/tree/4.1/media-manager

**Comment by LukasHH** [comment]
Is it planned that you can set up and authorize special subdirectories?

My Idea:
images\users\{username} =>authorized only for the respective user
images\usergroups\{groupname} =>authorized only for the respective user group
images\xyz =>authorized for user-defined user groups only

Also to the image selection for introduction and post, when posts are created, that one gets only the directories displayed, for which one is authorized.

**Reply by dgrammatiko** [reply]
> Is it planned that you can set up and authorize special subdirectories?

This is extremely easy to code: A plugin that registers a local adapter per user and according to config (and disables all other local adapters in the specific field)

**Comment by rinka88** [comment]
Isn't it better to take a ready-made quantum manager and rewrite for joomla 4
https://www.norrnext.com/quantum-manager

**Reply by bembelimen** [reply]
Most likely this will not happen, as we have a new media manager already with a lot of flexibility. Especially if some function is missing, it should be rather easy to implement them.

**Comment by dgrammatiko** [comment]
> and specifically playback of video and audio and support for documents.

Didn't I already introduce this functionality a few weeks ago?

Unfortunately, before jumping to provide new features, I guess the project should fix the list of pending issues: https://github.com/joomla/joomla-cms/discussions/34927 (it's not even the exhaustive list)

**Reply by bembelimen** [reply]
The playback of audio/video is here: https://github.com/joomla/joomla-cms/pull/35177 (I guess building on your code)

**Reply by brianteeman** [reply]
@bembelimen playback of audio/video is already present in the media manager without the need for any additional code.

**Reply by bembelimen** [reply]
Yes, sorry formulated it wrong, playing works already, there are potential improvements.

**Reply by dgrammatiko** [reply]
> there are potential improvements.

Actually, the preview has a bug right now and adding a JS player is not improving or solving the bug.
FWIW the bug goes like Preview a fairly large image, close the modal, now try to preview the next image. At the opening of the modal, you will be flashed with the old image (the same goes with audio/video/pdf). The solution is not complicated: the preview component needs to be async with the unresolved state as a spinner icon or something similar (eg loading...)
